app/scheduled/circulars/index.ts

```diff
@@ -0,0 +1,38 @@
+import { tables } from '@architect/functions'
+import type { DynamoDBDocument } from '@aws-sdk/lib-dynamodb'
+import { paginateScan } from '@aws-sdk/lib-dynamodb'
+
+import { uploadTar } from './uploadTar'
+import { type Circular } from '~/routes/circulars/circulars.lib'
+
+interface CircularAction {
+  callback: (circularArray: Circular[]) => void
+}
+
+async function mapCirculars(...actions: CircularAction[]) {
+  for await (const circularArray of getAllRecords()) {
+    for (const action of actions) {
+      action.callback(circularArray)
+    }
+  }
+}
+
+async function* getAllRecords(): AsyncGenerator<Circular[], void, unknown> {
+  const db = await tables()
+  const client = db._doc as unknown as DynamoDBDocument
+  const TableName = db.name('circulars')
+  const pages = paginateScan({ client }, { TableName })
+
+  for await (const page of pages) {
+    const items: Circular[] = page.Items as Circular[]
+    yield items
+  }
+}
+
+// FIXME: must use module.exports here for OpenTelemetry shim to work correctly.
+// See https://dev.to/heymarkkop/how-to-solve-cannot-redefine-property-handler-on-aws-lambda-3j67
+module.exports.handler = async () => {
+  const actions: CircularAction[] = [{ callback: uploadTar }]
```

Comment:
you put your actions in here after importing them. all that should need to be passed in is the array of circulars that are coming back in chunks.